packages/swapper/src/api.ts

```diff
@@ -55,18 +63,25 @@ export type CommonTradeInput = {
   wallet?: HDWallet // TODO remove this in a followup PR
 }
 
-export type GetEthTradeQuoteInput = CommonTradeInput & {
+export type GetCowTradeQuoteInput = CommonTradeInput & {
```

Comment:
I think we can just set the base Swapper interface to use the `CommonTradeInput` (maybe renamed to `GetTradeQuoteInput` and then each of these swapper specific extensions can be defined and used by that specific swapper instead of living in the top level api.

Reply:
I tried this in https://github.com/shapeshift/lib/pull/850/commits/3cda567995c8c29ab82a35176944d7311845960e, but then reverted it.

I think it's easier to let it discriminate automatically on `KnownChainIds`, then we can use the `GetTradeQuoteInput` type knowing that it'll find the right one for us in the union. 

A good example of this is in `setupSwapQuote.ts`, where we can keep `quoteInput` as `GetTradeQuoteInput` without needing to use the explicit return type of `GetEthereumTradeQuoteInput`, and then update all of the consumers etc.